src/NBitcoin/ConcurrentChain.cs

```diff
@@ -16,30 +16,29 @@ public class ConcurrentChain : ChainBase
         private ReaderWriterLock lockObject = new ReaderWriterLock();
 
         private volatile ChainedBlock tip;
+        private Network network;
         public override ChainedBlock Tip { get { return this.tip; } }
-
         public override int Height { get { return this.Tip.Height; } }
-
+        public override Network Network { get { return this.network; } }
+        
         public ConcurrentChain()
         {
+            this.network = Network.Main;
         }
-
-        public ConcurrentChain(BlockHeader genesisHeader)
+        
+        public ConcurrentChain(BlockHeader genesisHeader, Network network = null)
```

Comment:
Ok this has many places where its called but we have to be very careful here not to pass the wrong Network.
I would even say add a todo  to remove the `network = null` choice .

Reply:
Done